packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -73,13 +73,16 @@ function createHttpClient(): HttpClient {
 
 export function getCacheVersion(
   paths: string[],
-  compressionMethod?: CompressionMethod
+  compressionMethod?: CompressionMethod,
+  enableCrossOsArchive = false
 ): string {
-  const components = paths.concat(
-    !compressionMethod || compressionMethod === CompressionMethod.Gzip
-      ? []
-      : [compressionMethod]
-  )
+  const components = paths
+    .concat([!compressionMethod ? '' : compressionMethod])
+    .concat(
+      process.platform !== 'win32' || enableCrossOsArchive
+        ? []
+        : ['windows-only']
+    ) // Only check for windows platforms if enableCrossOsArchive is false
```

Comment:
Umm, can we push to `const`?

Reply:
We are doing so in [L#88](https://github.com/actions/toolkit/blob/4679d0cfe64af09ec136b2467b5fdc1a81fdf25d/packages/cache/src/internal/cacheHttpClient.ts#L88) too. I guess it would be copying the whole thing into a new array then.

Reply:
Right. But why is it better?

Reply:
Readability mostly.  `concat` is used to merge arrays whereas `push` is used to push single items onto it.  While reading through it made sense that concat was creating the components and then we are adding items to it using push. Although `concat` also has performance implications and `push` is generally faster.

Reply:
Got it. Added